sync/class.jetpack-sync-defaults.php

```diff
@@ -257,12 +257,22 @@ static function is_whitelisted_option( $option ) {
 		return false;
 	}
 
+	static function get_max_sync_execution_time() {
+		$max_exec_time = intval( ini_get( 'max_execution_time' ) );
+		if ( $max_exec_time === 0 ) {
+			// 0 actually means "unlimited", but let's not treat it that way
+		    $max_exec_time = 60;
```

Comment:
I thought about that, and tried a few different things. Basically, if there's ANY number returned above 0, it means use that number. If it returns 0, it ALWAYS means "no limit" (i.e. if the limit is really 30, it'll return 30).

So the function of this line is not to set it to the default if there's no value, it's to set it to a large-ish value if the config is set to unlimited. In theory we could also just execute for an unlimited amount of time ourselves, but it's better for reliability to periodically checkpoint and continue in a separate process.